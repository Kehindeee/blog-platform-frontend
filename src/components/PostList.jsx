import React from 'react';
import { Link } from 'react-router-dom';

const PostList = ({ posts, loading }) => {
  const formatDate = (dateString) => {
    const options = { year: 'numeric', month: 'long', day: 'numeric', hour: '2-digit', minute: '2-digit' };
    return new Date(dateString).toLocaleDateString(undefined, options);
  };

  if (loading) {
    return <p className="text-center">Loading...</p>;
  }

  return (
    <div className="space-y-8">
      {posts.map(post => (
        <div key={post.id} className="bg-white rounded-lg shadow-lg p-5 hover:shadow-2xl transition duration-300 ease-in-out">
          <h2 className="text-2xl font-semibold">
            <Link to={`/posts/${post.id}`}>{post.title}</Link> {/* Enable navigation to PostDetail */}
          </h2>
          <p className="text-gray-700">{post.content}</p>
          <div className="flex justify-between items-center">
            <p className="text-gray-500 text-sm">{formatDate(post.created_at)}</p>
            {/* Link to the post detail with a comment count (assuming 'commentCount' is a property of the post object) */}
            <Link to={`/posts/${post.id}`} className="text-gray-600 hover:underline">
              {post.commentCount ? `${post.commentCount} Comments` : 'No comments yet'}
            </Link>
          </div>
        </div>
      ))}
    </div>
  );
};

export default PostList;
