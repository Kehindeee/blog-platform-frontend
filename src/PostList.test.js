import React from 'react';
import { render, screen } from '@testing-library/react';
import PostList from './components/PostList';
import '@testing-library/jest-dom';
import { MemoryRouter } from 'react-router-dom'; // Import MemoryRouter

// Mock posts data for testing
const mockPosts = [
  {
    id: 1,
    title: 'First Post',
    content: 'This is the first post content',
    created_at: '2021-01-01T12:00:00Z',
    commentCount: 2,
  },
  {
    id: 2,
    title: 'Second Post',
    content: 'This is the second post content',
    created_at: '2021-01-02T12:00:00Z',
    commentCount: 3,
  },
];

describe('PostList Component', () => {
  it('renders posts when not loading', async () => {
    render(
      <MemoryRouter> {/* Wrap PostList in MemoryRouter for testing */}
        <PostList posts={mockPosts} loading={false} />
      </MemoryRouter>
    );

    // Check if the titles of the posts are rendered
    const firstPostTitle = screen.getByText('First Post');
    expect(firstPostTitle).toBeInTheDocument();

    const secondPostTitle = screen.getByText('Second Post');
    expect(secondPostTitle).toBeInTheDocument();
  });

  it('displays loading message when loading is true', () => {
    render(
      <MemoryRouter> {/* Wrap PostList in MemoryRouter for testing */}
        <PostList posts={[]} loading={true} />
      </MemoryRouter>
    );
    expect(screen.getByText('Loading...')).toBeInTheDocument();
  });
});
