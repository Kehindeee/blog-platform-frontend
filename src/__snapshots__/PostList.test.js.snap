// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PostList Component matches the snapshot 1`] = `
<div
  className="space-y-8"
>
  <div
    className="bg-white rounded-lg shadow-lg p-5 hover:shadow-2xl transition duration-300 ease-in-out"
  >
    <h2
      className="text-2xl font-semibold"
    >
      <a
        href="/posts/1"
        onClick={[Function]}
      >
        First Post
      </a>
       
    </h2>
    <p
      className="text-gray-700"
    >
      This is the first post content
    </p>
    <div
      className="flex justify-between items-center"
    >
      <p
        className="text-gray-500 text-sm"
      >
        1 January 2021 at 12:00
      </p>
      <a
        className="text-gray-600 hover:underline"
        href="/posts/1"
        onClick={[Function]}
      >
        2 Comments
      </a>
    </div>
  </div>
  <div
    className="bg-white rounded-lg shadow-lg p-5 hover:shadow-2xl transition duration-300 ease-in-out"
  >
    <h2
      className="text-2xl font-semibold"
    >
      <a
        href="/posts/2"
        onClick={[Function]}
      >
        Second Post
      </a>
       
    </h2>
    <p
      className="text-gray-700"
    >
      This is the second post content
    </p>
    <div
      className="flex justify-between items-center"
    >
      <p
        className="text-gray-500 text-sm"
      >
        2 January 2021 at 12:00
      </p>
      <a
        className="text-gray-600 hover:underline"
        href="/posts/2"
        onClick={[Function]}
      >
        3 Comments
      </a>
    </div>
  </div>
</div>
`;
